soda/scientific/soda/scientific/distribution/generate_dro.py

```diff
@@ -16,14 +17,51 @@ class Config:
         arbitrary_types_allowed = True
 
 
-def normalize(data: np.ndarray):
+def normalize(data: np.ndarray) -> np.ndarray:
     return data / np.sum(data)
 
 
 class DROGenerator:
     def __init__(self, cfg: RefDataCfg, data: list) -> None:
         self.distribution_type = cfg.distribution_type
         self.data = data
+        self.maximum_allowed_bin_size = 1e6
+
+    @staticmethod
+    def _compute_n_bins(data: np.ndarray) -> int:
+        _range = (data.min(), data.max())
+        bin_width = np.lib.histograms._hist_bin_auto(data, _range)
+        first_edge, last_edge = np.lib.histograms._get_outer_edges(data, _range)
+        n_bins = int(np.ceil(np.lib.histograms._unsigned_subtract(last_edge, first_edge) / bin_width))
+        return n_bins
+
+    @staticmethod
+    def _remove_outliers_with_iqr(data: np.ndarray) -> np.ndarray:
+        # Remove outliers
+        q1, q3 = np.percentile(data, [25, 75])
+        IQR = q3 - q1
+        lower_range = q1 - (1.5 * IQR)
+        upper_range = q3 + (1.5 * IQR)
+        filtered_data = data[np.where((data >= lower_range) & (data <= upper_range))]
+        logging.warning(
+            f"""While generating the distribution reference object, the automatic bin size detection
```

Comment:
Here I would rephrase a bit:

"Generating the distribution reference object using automatic bin size detection would cause a memory error. This is generally cause by the presence of outliers in the dataset which leads to <INSERT_REASON>."

I think we might want to cut out the part where we give the code, and instead make a PR to the docs repo where we could give some details about this. And instead, provide a link to said docs page.

Then continue with "We filtered values above X and below Y using IQR and provide the link."

Then I would also make sure we have a more verbose explanation of this in the docs PR so that we can also add a link here.